### PR TITLE
Simplify DynamicTransformer API

### PR DESCRIPTION
**Pull request by felixbarny**
## What does this PR do?

Instead of `DynamicTransformer.Accessor.get().ensureInstrumented`,users can just call `DynamicTransformer.ensureInstrumented`

Makes the API similar in style to the recent changes in `WeakConcurrent`.

## Checklist

- [ ] This is something else
  - [x] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/master/CHANGELOG.asciidoc)

### CODE REVIEW

File: apm-agent-plugin-sdk/src/main/java/co/elastic/apm/agent/sdk/DynamicTransformer.java [hunk 1]
```diff
@@ -38,23 +48,13 @@
...
-        }
-
-        public static DynamicTransformer get() {
-            return transformer;
-        }
+    public static void ensureInstrumented(Class<?> classToInstrument, Collection<Class<? extends ElasticApmInstrumentation>> instrumentationClasses) {
+        transformer.ensureInstrumented(classToInstrument, instrumentationClasses);
     }
 
+    public interface DynamicTransformerSupplier {
```
Comment (eyalkoren) [inline apm-agent-plugin-sdk/src/main/java/co/elastic/apm/agent/sdk/DynamicTransformer.java:55]: [minor] I don't think `DynamicTransformerSupplier` is a good name for this interface. I would call it `DynamicTransformer` and find a better name for the declaring class, even `StaticDynamicTransformer`, but preferably something better 🙂

**Review by eyalkoren — APPROVED** [review]
LGTM, only a minor naming suggestion

File: apm-agent-plugin-sdk/src/main/java/co/elastic/apm/agent/sdk/DynamicTransformer.java (hunk 1 above)
Comment (felixbarny) [reply apm-agent-plugin-sdk/src/main/java/co/elastic/apm/agent/sdk/DynamicTransformer.java:55]: I agree that the name of the interface isn't ideal. But I wanted to optimize for the actual user facing class which is `DynamicTransformer`. While `DynamicTransformerSupplier` needs to be public, users never really get to see it. Also, it's in line with `WeakConcurrentSupplier`.

Comment (eyalkoren) [reply apm-agent-plugin-sdk/src/main/java/co/elastic/apm/agent/sdk/DynamicTransformer.java:55]: It's just that `DynamicTransformer` doesn't supply anything (as opposed to the weak map supplier).
What about `RuntimeInstrumentor` for the interface?

And an additional suggestion - I just noticed that `DynamicTransformerImpl` seems redundant, why not make `ElasticApmAgent` the service provider implementing the interface (whatever it's called 🙂  ).

Either way, feel free to ignore both suggestions and merge if you disagree. It's not that important.

**Pull request merged**